quotefault/__init__.py

```diff
@@ -215,28 +237,138 @@ def get():
+    quotes = get_quote_query(speaker = request.args.get('speaker'),
+        submitter = request.args.get('submitter')).all()
 
     #tie any votes the user has made to their uid
     user_votes = db.session.query(Vote).filter(Vote.voter == metadata['uid']).all()
 
-
     return render_template(
         'bootstrap/additional_quotes.html',
         quotes=quotes[20:],
         metadata=metadata,
         user_votes=user_votes
     )
+
+@app.route('/report/<report_id>', methods=['POST'])
+@auth.oidc_auth
+def submit_report(report_id):
+    """
+    Report a quote and notify EBoard/RTP
+    """
+    metadata = get_metadata()
+    existing_report = Report.query.filter(Report.reporter==metadata['uid'],
+        Report.quote_id==report_id).first()
+    if existing_report:
+        flash("You already submitted a report for this Quote!")
+        return redirect('/storage')
+    new_report = Report(report_id, metadata['uid'], None)
+    db.session.add(new_report)
+    db.session.commit()
+    if app.config['MAIL_SERVER'] != '':
+        send_report_email( metadata['uid'], Quote.query.get(report_id) )
+
+    flash("Report Successful!")
+    return redirect('/storage')
+
+@app.route('/review', methods=['GET'])
+@auth.oidc_auth
+def review():
+    """
+    Presents page for admins to review reports
+    and either hide or keep quote
+    """
+    metadata = get_metadata()
+    if metadata['is_admin']:
+        # Get the most recent 20 quotes
+        reports = Report.query.filter(Report.reviewed == False).all()
+
+        return render_template(
+            'bootstrap/admin.html',
+            reports=reports,
+            metadata=metadata
+        )
+    return redirect('/')
+
+@app.route('/review/<report_id>/<result>', methods=['POST'])
+@auth.oidc_auth
+def review_submit(report_id, result):
+    """
+    Called when Admin decides on a quote being hidden or kept
+    """
+    metadata = get_metadata()
+    result = int(result)
+    if metadata['is_admin']:
+        # 1 = Keep, 0 = Hide
+        if result:
+            report = Report.query.get(report_id)
+            report.reviewed = True
+            db.session.commit()
+            flash("Report Completed: Quote Kept")
+        else:
+            report = Report.query.get(report_id)
+            report.quote.hidden = True
+            report.reviewed = True
+            db.session.commit()
+            flash("Report Completed: Quote Hidden")
+        return redirect('/review')
+    return redirect('/')
+
+@app.route('/hide/<report_id>', methods=['POST'])
+@auth.oidc_auth
+def hide(report_id):
+    """
+    Hides a quote
+    """
+    metadata = get_metadata()
+    quote = Quote.query.get(report_id)
+    if ( metadata['uid'] == quote.speaker
+        or metadata['uid'] == quote.submitter
+        or metadata['is_admin'] ):
+        quote.hidden = True
+        db.session.commit()
+        flash("Quote Hidden!")
+    return redirect('/storage')
+
+@app.route('/unhide/<report_id>', methods=['POST'])

```

Comment:
same questions about redirects, plus you should return an error if the user isn't allowed.